protoutil/conversion.go

```diff
@@ -39,18 +40,13 @@ func ToProtoStruct(v any) (*structpb.Struct, error) {
 	return x, nil
 }
 
-// UnmarshalYAML unmarshals the give YAML bytes to the given proto message.
-func UnmarshalYAML(b []byte, msg proto.Message) error {
-	tmp := map[string]any{}
-	if err := yaml.Unmarshal(b, tmp); err != nil {
-		return fmt.Errorf("failed to unmarshal yaml: %w", err)
-	}
-	jb, err := json.Marshal(tmp)
+// YAMLToProto converts YAML to Proto.
+func YAMLToProto(y []byte, m proto.Message) error {
```

Comment:
I feel renaming makes the functionality more clear. I can search the existing usage of "UnmarshalYAML" and change it to "YAMLToProto". Let me know if there is any concerns.

Reply:
I think it makes sense. UnmarshalYAML makes me think its more general purpose.

Reply:
The package name is `protoutil`. `protoutil.UnmarshalYAML` seems clear enough?

Reply:
As Seth suggested, I changed it to 'FromYAML', is that ok?